login_page.php

```diff
@@ -141,10 +141,29 @@
 $t_upgrade_required = false;
 
 if( config_get_global( 'admin_checks' ) == ON ) {
-	# Check if the admin directory is accessible
+	# Check if admin directory is accessible at filesystem level
 	$t_admin_dir = dirname( __FILE__ ) . '/admin';
 	$t_admin_dir_is_accessible = @file_exists( $t_admin_dir . '/.' );
-	if( $t_admin_dir_is_accessible ) {
+
+	# Check if admin directory is accessible as URL by retrieving HTTP headers.
+	try {
+		$t_guzzle = new GuzzleHttp\Client( array(
+			'base_uri' => config_get_global( 'path' ),
+			'timeout' => 1,
```

Comment:
Default timeout is 0 (no timeout). I was thinking about avoiding unnecessary delay for what is basically a non-essential function. But considering it's basically Mantis accessing itself, on second thoughts it is probably not necessary.